test/unit/ngsiv2/MQTT_receive_measures-test3.js

```diff
@@ -0,0 +1,140 @@
+                apikey: 'KL223HHV8732SFL1',
+                entity_type: 'TheLightType',
+                trust: '8970A9078A803H3BL98PINEQRW8342HBAMS',
+                cbHost: 'http://192.168.1.1:1026',
+                commands: [],
+                lazy: [],
+                attributes: [
+                    {
+                        name: 'status',
+                        type: 'Boolean'
+                    }
+                ],
+                static_attributes: []
+            }
+        ]
+    },
+    headers: {
+        'fiware-service': 'smartgondor',
+        'fiware-servicepath': '/gardens'
+    }
+};
+let contextBrokerMock;
+let contextBrokerUnprovMock;
+let mqttClient;
+
+describe('MQTT: Measure reception ', function () {
+    beforeEach(function (done) {
+        const provisionOptions = {
+            url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDevice1.json'),
+            headers: {
+                'fiware-service': 'smartgondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        mqttClient = mqtt.connect('mqtt://' + config.mqtt.host, {
+            keepalive: 0,
+            connectTimeout: 60 * 60 * 1000
+        });
+
+        // This mock does not check the payload since the aim of the test is not to verify
+        // device provisioning functionality. Appropriate verification is done in tests under
+        // provisioning folder of iotagent-node-lib
+        contextBrokerMock = nock('http://192.168.1.1:1026')
+            .matchHeader('fiware-service', 'smartgondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v2/entities?options=upsert')
+            .reply(204);
+
+        iotaJson.start(config, function () {
+            request(provisionOptions, function (error, response, body) {
+                done();
+            });
+        });
+    });
+
+    afterEach(function (done) {
+        nock.cleanAll();
+        mqttClient.end();
+
+        async.series([iotAgentLib.clearAll, iotaJson.stop], done);
+    });
+
+    describe('When a POST single Raw measure arrives for the HTTP binding', function () {
+        beforeEach(function () {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartgondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .patch(
+                    '/v2/entities/Second%20MQTT%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasuresRawTypes1.json')
+                )
+                .query({ type: 'AnMQTTDevice' })
+                .reply(204);
+        });
+        it('should send its value to the Context Broker', function (done) {
+            mqttClient.publish('/1234/MQTT_2/attrs/humidity', '0x4D', null, function (error) {
```

Comment:
```suggestion
            mqttClient.publish('/1234/MQTT_2/attrs/humidity', '0x4D', null, function (error) {
```
This case is not totally correct. You are not sending binary data through MQTT here. When sending '0x4D', you are sending a **string of ascii chars**. In that case, the binary representation (in Hex) is made of 4 bytes:

**Hex representation**
```
0x30,0x78,0x34,0x44
```

**Binary representation**
```
00110000 01111000 00110100 01000100
```
So, the expected value persisted in that case should be:
```json
"value": "30783444"
```

So, the initial part (0x), should not be removed

In order to don't lead to confusion, I suggest to change the test string without including the initial part `0x`. 
Additionally, the part of the code that removes `0x` should be removed